cmd/frontend/internal/app/ui/sveltekit/gen/gen_routes.go

```diff
@@ -0,0 +1,303 @@
+	writeTSFile(filepath.Join(*dest, "_routes.ts"), routes)
+}
+
+func writeGoFile(dest string, routes []routeInfo) error {
+	t := template.Must(template.New("routes").Funcs(template.FuncMap{
+		"joinTags": func(tags []string, sep string) string {
+			if len(tags) == 0 {
+				return ""
+			}
+			return "tags." + strings.Join(tags, " | tags.")
+		},
+	}).Parse(goFileTemplate))
+
+	// Write template to _routes.go file
+	file, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return t.Execute(file, map[string]interface{}{
+		"Routes": routes,
+	})
+}
+
+func writeTSFile(dest string, routes []routeInfo) error {
+	t := template.Must(template.New("routes").Funcs(template.FuncMap{
+		"isRepoRoot": func(route routeInfo) bool {
+			for _, tag := range route.Tags {
+				if tag == "RepoRoot" {
+					return true
+				}
+			}
+			return false
+			},
+		}).Parse(tsFileTemplate))
+
+	// Write template to _routes.ts file
+	file, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	return t.Execute(file, map[string]interface{}{
+		"Routes": routes,
+	})
+}
+
+func getPagePaths() []string {
+	var paths []string
+
+	// When running under Bazel, the page file paths are passed as arguments.
+	args := flag.Args()
+	if len(args) > 0 {
+		return args
+	}
+
+	err := filepath.Walk(SRC_ROUTES_PREFIX, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if filepath.Base(path) == "+page.svelte" {
+			paths = append(paths, path)
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error walking directory: %v", err)
+		os.Exit(1)
+	}
+	return paths
+}
+
+var (
+	splitPattern = regexp.MustCompile(`\s+`)
+	tagPattern = regexp.MustCompile(`^\s*//\s+@sg\s+`)
+	groupPattern = regexp.MustCompile(`^\([^)]+\)$`)
+	paramPattern = regexp.MustCompile(`^(\[)?(\.\.\.)?(\w+)(?:=(\w+))?(\])?$`)
+	restParamPattern = regexp.MustCompile(`^\[\.\.\.(\w+)(?:=(\w+))?\]$`)
+	optionalParamPattern = regexp.MustCompile(`^\[\[(\w+)(?:=(\w+))?\]\]$`)
+)
+
+// extractTags finds lines of the forms
+//     // @sg tag1 tag2 tag3
+// and returns the tags found.
+func getRouteInfo(path string) (routeInfo, error) {
+	var info routeInfo
+
+	f, err := os.Open(path)
+	if err != nil {
+		return info, err
+	}
+	defer f.Close()
+
+	// All routes are opt-in by default
+	tagNames := []string{"EnableOptIn"}
+
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if tagPattern.MatchString(line) {
+			match := tagPattern.FindString(line)
+			tagNames = append(tagNames, splitPattern.Split(strings.TrimPrefix(line, match), -1)...)
+			break
+		}
+	}
+
+	for _, tag := range tagNames {
+		if !tags.IsTagValid(tag) {
+			return info, errors.Newf("Invalid tag '%s'. Valid tags: %s", tag, tags.AvailableTags())
+		}
+	}
+	info.Tags = tagNames
+
+	if err := scanner.Err(); err != nil {
+		return info, err
+	}
+
+	return info, nil
```

Comment:
style nit: I generally prefer returning a fully-constructed object rather than mutating a local variable and returning it. It just makes it a bit easier to tell what exactly is in the object being returned.

```suggestion

	if err := scanner.Err(); err != nil {
		return info, err
	}

	return routeInfo{Tags: tagNames}, nil
```